swap/log.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	log "github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethersphere/swarm/network"
+)
+
+const (
+	// UndefinedAction is the default actions filter for swap logs
+	UndefinedAction string = "undefined"
+	// InitAction used when starting swap
+	InitAction string = "init"
+	// StopAction used when stopping swap
+	StopAction string = "stop"
+	// UpdateBalanceAction used when updating balances
+	UpdateBalanceAction string = "update_balance"
+	// SendChequeAction used for cheque actions
+	SendChequeAction string = "send_cheque"
+	// HandleChequeAction used for cheque actions
+	HandleChequeAction string = "handle_cheque"
+	// CashChequeAction used for cheque actions
+	CashChequeAction string = "cash_cheque"
+	// DeployChequebookAction used when deploying chequebooks
+	DeployChequebookAction string = "deploy_chequebook_contract"
+)
+
+const swapLogLevel = 3       // swapLogLevel indicates filter level of log messages
```

Comment:
You could do:
`log.Lvl(ctx.Int("verbosity"))`

Reply:
To use `ctx.Int("verbosity")`
`ctx *cli.Context` should be passed into the logger.

This would not be an issue, but it would be pegged to the log level on init.

This const was defined explicitly to always log swap at a certain level, since it's purpose is auditing swap related logs.

Maybe a cli flag could be passed for swap log level filter, I did not want to clutter the cli.
But this a valid option as well.

Do you think this could be a valid approach?.